### ISSUE TITLE
Test new sortable icon markup

### ISSUE THREAD

**Issue by kimisgold** [omeka/plugin-ItemOrder#10]
We're revising Omeka Classic's collapsible drawer pattern and applying it to the plugins. This pattern includes components that can collapse and expand, as well as adjustments to how we mark up sortable elements. This issue centers the item order interface on the collection show view. We mainly want to check that sortable icon and all the item content is readable by screen readers. We're not looking at the actual sorting functionality's accessibility, which is being addressed in a separate effort.

You need to be on the `drawer-toggle-a11y-rebase` branch for Omeka Classic, and the `drawer-toggle-a11y` branch of ItemOrder.

**Comment by aviannamiller** [comment]
Just providing some preliminary feedback from testing with VO on desktop (will follow up with others):

- I am only seeing the move icons receive focus via arrow key navigation. They cannot be tabbed to, which is not inherently an issue but can be limiting. Is this because the icon is not a button or link and rather the element's actions are dictated by CSS?
- There is something illogical at play here with the sort icon's relationship to the listed item metadata. The move icon is read out as part of an empty group and does not provide any identification regarding a given item. The item title (then date of internal creation, then item page link) follows, but it is nearly impossible to tell via screen reader what you are moving (given that functionality is there) without having to navigate forward, then backward. Would any of the following make sense as a solution?:
  - Move the sort icons to the right hand side of their containers so item metadata is read out first
  - Label sort icons in a way that also includes the item title (maybe via `aria-labelledby`, though I am unsure if that works here to this effect)
  -  Give focus to the container of a list item before it can be entered into and provide the item title in the label in a similar fashion to the above suggestion, although there are resulting concerns for both of these of repetition in the read out
- As another note, the item page link could be more descriptive, either by enhancing the meaning of the link text or adding hidden link text / an appropriate ARIA attribute.

**Comment by aviannamiller** [comment]
- When focus is on the sort icon, TalkBack (on Android) reads out a list item's given location in X amount of list items (e.g. 1 of 3) instead of "Move".
- JAWS adds parentheses into the announcement for the date/time metadata and link as per how they appear in text. Not really a big issue, but for shorter metadata this adds a bit of unnecessary bulk in the read out. I am wondering if there is a different approach to styling the text for these elements to avoid this, but again, not really a huge concern.

Now fully tested with TalkBack, VoiceOver, NVDA, and Jaws.

**Comment by kimisgold** [comment]
> I am only seeing the move icons receive focus via arrow key navigation. They cannot be tabbed to, which is not inherently an issue but can be limiting. Is this because the icon is not a button or link and rather the element's actions are dictated by CSS?

Yes, exactly this. I'm wondering if it needs to be exposed to screen readers at all if it's an indicator to functionality inaccessible to keyboards, but I erred on the side of yes at the time. I'm now leaning towards setting it to `aria-hidden="true"`as it's only useful to sighted users. I've pushed this change.

> As another note, the item page link could be more descriptive, either by enhancing the meaning of the link text or adding hidden link text / an appropriate ARIA attribute.

I've made the item link label more descriptive, as well as justified it right and added an external link icon since it opens into a new window. 

The list item container for each item should now have `role="group"`, labelled by the item title. Screen readers should narrate that out as soon as they move into the item contents.

**Comment by aviannamiller** [comment]
> Yes, exactly this. I'm wondering if it needs to be exposed to screen readers at all if it's an indicator to functionality inaccessible to keyboards, but I erred on the side of yes at the time. I'm now leaning towards setting it to `aria-hidden="true"`as it's only useful to sighted users. I've pushed this change.

Agreed. This seems like a good choice for the time being.
> I've made the item link label more descriptive, as well as justified it right and added an external link icon since it opens into a new window.

This is great, thanks!

> The list item container for each item should now have `role="group"`, labelled by the item title. Screen readers should narrate that out as soon as they move into the item contents.

It is my understanding that this type of [ARIA role is not allowed on this kind of element](https://dequeuniversity.com/rules/axe/4.3/aria-allowed-role), although I do not see any issues (retested across screen readers and devices) being triggered by this at this point.

**Comment by kimisgold** [comment]
Easy enough to remove the group role from the list item. I've also used the item title for the links' `aria-describedby` attributes to better associate them without the group container.

**Comment by aviannamiller** [comment]
Great, thank you! I'm now getting a helpful read out on those links on all tested except VO, although it looks like we can chalk that up to the [reported partial support of this attribute](https://a11ysupport.io/tech/aria/aria-describedby_attribute).

**Comment by kimisgold** [comment]
The partial support is kind of surprising, and I don't see a better alternative. Hopefully context with the link being so near the item title is enough without that support. Is this good to close otherwise?

**Comment by aviannamiller** [comment]
Definitely. It's okay to close for now; let's just keep this in mind for future use and possible revisions.
